core/arch/arm/mm/tee_mmu.c

```diff
@@ -263,7 +263,7 @@ TEE_Result tee_mmu_init(struct tee_ta_ctx *ctx)
 TEE_Result tee_mmu_map(struct tee_ta_ctx *ctx, struct tee_ta_param *param)
 {
 	TEE_Result res = TEE_SUCCESS;
-	paddr_t pa;
+	paddr_t pa = NULL;
```

Comment:
@jenswi-linaro you cannot assume GCC will generate correct code to access address 0 if you don't add `-fno-delete-null-pointer-checks`, I learned it when I switched to GCC 4.9 to build ARM Trusted Firmware for HiKey. See this thread: https://lists.96boards.org/pipermail/dev/2015-March/000138.html and especially this message: https://lists.96boards.org/pipermail/dev/2015-March/000150.html.